src/main/scala/esmeta/analyzer/domain/state/TypeDomain.scala

```diff
@@ -296,26 +300,39 @@ object TypeDomain extends state.Domain {
     if (str contains "Type")
       if (normal) res ||= CONSTT_NORMAL
       if (abrupt) res ||= constTyForAbruptTarget
-    // TODO if (!comp.isBottom)
-    //   boundCheck(
-    //     prop,
-    //     StrT("Value", "Target", "Type"),
-    //     t => s"invalid access: $t of $comp",
-    //   )
+    if (!comp.isBottom)
+      boundCheck(
+        prop,
+        StrT("Value", "Target", "Type"),
+        (cp, t) =>
+          val plp = PropertyLookupPoint(LookupKind.Str, cp, ref)
+          UnknownPropertyMismatch(plp, comp, t),
+      )
     res
 
   // AST lookup
-  private def lookupAst(ast: AstValueTy, prop: ValueTy): ValueTy = ast match
-    case AstValueTy.Bot => ValueTy.Bot
-    case AstSingleTy(name, idx, subIdx) =>
-      lookupAstIdxProp(name, idx, subIdx)(prop) ||
-      lookupAstStrProp(prop)
-    case AstNameTy(names) =>
-      if (!prop.math.isBottom) AstT // TODO more precise
-      else lookupAstStrProp(prop)
-    case _ => AstT
-  // TODO if (!ast.isBottom)
-  //   boundCheck(prop, MathT || StrT, t => s"invalid access: $t of $ast")
+  private def lookupAst(
+    ast: AstValueTy,
+    prop: ValueTy,
+    ref: Option[Ref],
+  ): ValueTy =
+    if (!ast.isBottom)
```

Comment:
Is there any reason why boundCheck comes at the top of the function in this case?

Reply:
I just want to preserve implicit return